src/uct/ib/base/ib_verbs.h

```diff
@@ -171,5 +171,18 @@ static inline int ibv_exp_cq_ignore_overrun(struct ibv_cq *cq)
 #  define IBV_EXP_HAVE_ATOMIC_HCA_REPLY_BE(_attr)   0
 #endif /* HAVE_DECL_IBV_EXP_ATOMIC_HCA_REPLY_BE */
 
+static inline void uct_ib_qpnum2buf(uint8_t buf[3], const uint32_t qp_num)
```

Comment:
i think it would be nicer to do something like 
`uct_ib_qpnum_pack(uct_ib_qpnum_t *buf, uint32_t qp_num)`
and then define rc iface address, and ud iface address, as uct_ib_qpnum_t (replace uct_rc_ep_addr_t)
and also embed this struct in ud ep address